indico/modules/events/requests/__init__.py

```diff
@@ -49,7 +50,10 @@ def _merge_users(target, source, **kwargs):
 @signals.event.deleted.connect
 def _event_deleted(event, **kwargs):
     from indico.modules.events.requests.models.requests import Request, RequestState
+
+    is_past_event = event.start_dt < now_utc()
```

Comment:
An ongoing event isn't really "past", so better to check using the end time:

```suggestion
    is_past_event = event.end_dt < now_utc()
```